mne/epochs.py

```diff
@@ -1923,7 +1924,9 @@ def equalize_epoch_counts(epochs_list, method='mintime'):
         raise ValueError('All inputs must be Epochs instances')
 
     # make sure bad epochs are dropped
-    [e.drop_bad_epochs() if not e._bad_dropped else None for e in epochs_list]
+    for e in epochs_list:
+        if not e._bad_dropped:
+            e.drop_bad_epochs()
```

Comment:
Also what I don't like about it is the fact that at least in python 3 the list comprehension would have created a new scope, preventing `e` from being available outside of it. Not sure if the codacy advice is so brilliant here. Exchanging such a simple list comprehension for a foor loop does not seem to be a good deal to me.